react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faBars } from '@fortawesome/free-solid-svg-icons';
+
+class AppDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isOpen: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+    if (event.target.className !== 'menu' && event.target.className !== 'rectangle') {
+      this.setState({ isOpen: !this.state.isOpen });
+    }
```

Comment:
Rather than specifying what the event target _isn't_ (which could be a lot of things), check whether the `target` (the element the user clicked on) is the same as the `currentTarget` (the element with the event listener): https://reactjs.org/docs/events.html